### PR TITLE
Bump native PlatformToolset to v143

### PR DESCRIPTION
**Pull request by RassK**
### What

Updates native platform toolset to the latest version.
(proposed by VS2022)

### CODE REVIEW

**Comment by RassK** [comment]
Seems it can't find v143 somehow, although VS2022 is used and windows-2022 runner 🤔

**Comment by pellared** [comment]
> Seems it can't find v143 somehow, although VS2022 is used and windows-2022 runner 🤔

Does the runner have the newest VS2002? https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#visual-studio-enterprise-2022

**Comment by RassK** [comment]
> > Seems it can't find v143 somehow, although VS2022 is used and windows-2022 runner 🤔
> 
> Does the runner have the newest VS2022? https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#visual-studio-enterprise-2022

I was also looking that list and in the same time the GHA issue here > 
https://github.com/actions/virtual-environments/issues/4468

Seems something has gone wrong and v143 is still not installed.

**Comment by pellared** [comment]
> I was also looking that list and in the same time the GHA issue here > [actions/virtual-environments#4468](https://github.com/actions/virtual-environments/issues/4468)
> 
> Seems something has gone wrong and v143 is still not installed.


I think that [`microsoft/setup-msbuild`](https://github.com/microsoft/setup-msbuild) GH action has to be used.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
         machine: [ windows-2022, ubuntu-20.04, macos-10.15 ]
     runs-on: ${{ matrix.machine }}
     steps:
+      - uses: microsoft/setup-msbuild@v1.1
```
Comment (Kielek) [inline .github/workflows/ci.yml:17]:
```suggestion
      - uses: microsoft/setup-msbuild@v1.1.0
```

Comment (pellared) [reply .github/workflows/ci.yml]: Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/401/commits/86cd11714e739c284d26809bff56f1cbff51a7af

Comment (pellared) [reply .github/workflows/ci.yml:17]: It occurred to be good... https://github.com/microsoft/setup-msbuild/releases/tag/v1.1

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@RassK It was required to bump https://www.nuget.org/packages/Microsoft.googletest.v140.windesktop.msvcstl.static.rt-static/ . Found it here https://developercommunity.visualstudio.com/t/issue-with-googletest-project-when-upgrading-to-vs/1576499

**Review by Kielek — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**